clang-tools-extra/clang-tidy/modernize/MinMaxUseInitializerListCheck.cpp

```diff
@@ -0,0 +1,337 @@
+            Result.Compare = *ArgIterator;
+          }
+          return Result;
+        }
+      }
+    }
+  }
+
+  for (const Expr *Arg : Call->arguments()) {
+    if (!Result.First)
+      Result.First = Arg;
+
+    if (Arg == Result.Compare)
+      continue;
+
+    Result.Args.push_back(Arg);
+    Result.Last = Arg;
+  }
+
+  return Result;
+}
+
+static std::vector<std::pair<int, int>>
+getCommentRanges(const std::string &source) {
+  std::vector<std::pair<int, int>> commentRanges;
+  bool inComment = false;
+  bool singleLineComment = false;
+  int start = -1;
+  for (unsigned long i = 0; i < source.size(); i++) {
+    switch (source[i]) {
+    case '/':
+      if (source[i + 1] == '/')
+        singleLineComment = true;
+      else if (source[i + 1] == '*')
+        inComment = true;
+      break;
+    case '*':
+      if (source[i + 1] == '/')
+        inComment = false;
+      break;
+    case '\n':
+      singleLineComment = false;
+      break;
+    }
+    if (inComment || singleLineComment) {
+      if (start == -1)
+        start = i;
+    } else if (start != -1) {
+      commentRanges.push_back({start, i});
+      start = -1;
+    }
+  }
+  return commentRanges;
+}
+
+static bool
+isPositionInComment(int position,
+                    const std::vector<std::pair<int, int>> &commentRanges) {
+  return std::any_of(commentRanges.begin(), commentRanges.end(),
+                     [position](const auto &range) {
+                       return position >= range.first &&
+                              position <= range.second;
+                     });
+}
+
+static void
+removeCharacterFromSource(std::string &FunctionCallSource,
+                          const std::vector<std::pair<int, int>> &CommentRanges,
+                          char Character, const CallExpr *InnerCall,
+                          std::vector<FixItHint> &Result, bool ReverseSearch) {
+  size_t CurrentSearch = ReverseSearch ? FunctionCallSource.size() : 0;
+  while ((CurrentSearch =
+              ReverseSearch
+                  ? FunctionCallSource.rfind(Character, CurrentSearch - 1)
+                  : FunctionCallSource.find(Character, CurrentSearch + 1)) !=
+         std::string::npos) {
+    if (!isPositionInComment(CurrentSearch, CommentRanges)) {
+      Result.push_back(FixItHint::CreateRemoval(CharSourceRange::getCharRange(
+          InnerCall->getSourceRange().getBegin().getLocWithOffset(
+              CurrentSearch),
+          InnerCall->getSourceRange().getBegin().getLocWithOffset(
+              CurrentSearch + 1))));
+      break;
+    }
+  }
+}
+
+SourceLocation getLocForEndOfToken(const Expr *expr,
+                                   const MatchFinder::MatchResult &Match) {
+  return Lexer::getLocForEndOfToken(expr->getEndLoc(), 0, *Match.SourceManager,
+                                    Match.Context->getLangOpts());
+}
+
+static const std::vector<FixItHint>
+generateReplacement(const MatchFinder::MatchResult &Match,
```

Comment:
this function is over-complicated.
You do not need to do any text parsing, checking characters and so on.
All you need to do is to replace anything that is bettwen arguments with `,` and replace anything bettwen beiging of top call and first argument with `std::min({` and last argument and end with `})`. Comments does not need to be preserved.

Reply:
I've rewritten this as a vector of `FixItHint`s specifically to preserve comments. Right now the only if statement that's there only to preserve comments is on on line 143-146

```cpp
if (utils::lexer::getPreviousToken(
                  InnerResult.Compare->getExprLoc(), *Match.SourceManager,
                  Match.Context->getLangOpts(), false)
                  .getLocation() == Comma->getLocation())
```

to know if there's any comment between the comma and the compare function and create an appropriate removal to preserve the it. I can simplify this and simply remove everything after the last argument in the case there is a compare function (deleting the comment as well), if you'd like me to.